tutorials/basic-topics/sorting/selection-sort.md

```diff
@@ -11,4 +11,110 @@ keywords:
   - selection sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@Bobliuuu"/>
+
+## Overview
+
+Selection sort is a commonly used comparison-based sorting algorithm, that is also very simple to implement. 
+Selection sort works on the premise that two subarrays are maintained: one which is **sorted**, and one which is **unsorted**. 
+Each step, one more element of the array gets sorted, until the entire array is sorted. 
+
+- **subarray**: A subsequence or slice of an array
+
+## Algorithm
+
+The concept of selection sort is that each time, we find the minimum element in the unsorted subarray, and we put it at the end of the sorted subarray. 
+Consider an array with n distinct elements. Looping n times, we
+- Find the minimum element in the unsorted subarray
+- Move that element to the end of the sorted subarray
```

Comment:
Suppose you've found the smallest number as an element, where would you move it ? First or last ?

Reply:
The minimum element in the unsorted is always the maximum in the sorted subarray, so we always move it to the end of the sorted subarray. 
Assume by contradiction the min element in the unsorted element doesn't go at the end. Then it must go somewhere else, but this cannot be possible because the end of the sorted subarray will always be smaller than it, since the minimum was achieved at a previous step.